plugins/cpp_reparse/service/src/cppreparseservice.cpp

```diff
@@ -144,6 +153,109 @@ void CppReparseServiceHandler::getAsHTMLForNode(
   return_ = htmlFactory.str();
 }
 
+void CppReparseServiceHandler::getSpecialMembersSource(
+  std::vector<language::SourceTextFragment>& return_,
+  const core::AstNodeId& astNode_)
+{
+  if (!isEnabled())
+  {
+    LOG(warning) << "Reparse capabilities has been disabled at server start.";
+    return;
+  }
+
+  // Fetch the AST node that that is either the record definition specified by
+  // the user, or the type definition for the class of the member function
+  // the user clicked on.
+  model::CppAstNodePtr astNode;
+  _transaction([&, this]() {
+    astNode = _db->query_one<model::CppAstNode>(
+      AstQuery::id == std::stoull(astNode_));
+    if (!astNode)
+      return;
+
+    if (astNode->symbolType == model::CppAstNode::SymbolType::Function)
+    {
+      auto memberType = _db->query_one<model::CppMemberType>(
+        MemTypeQuery::memberAstNode == astNode->id &&
+        MemTypeQuery::kind == model::CppMemberType::Kind::Method);
+      astNode = nullptr;
+      if (!memberType)
+      {
+        LOG(warning) << "Couldn't fetch this method as a member ... ?!";
+        return;
+      }
+
+      for (const model::CppAstNode& resultType : _db->query<model::CppAstNode>(
+        AstQuery::mangledNameHash == memberType->typeHash &&
+        AstQuery::symbolType == model::CppAstNode::SymbolType::Type &&
+        AstQuery::astType == model::CppAstNode::AstType::Definition))
+      {
+        astNode = std::make_shared<model::CppAstNode>(resultType);
+        break;
+      }
+
+      if (!astNode)
+      {
+        LOG(warning)
+          << "Couldn't find the type definition for the member function";
+        return;
+      }
+    }
+  });
+
+  if (!astNode)
+  {
+    LOG(warning) << "No location file found for AST node #" << astNode_;
+    return;
+  }
+
+  TypeSpecialMemberPrinter::DefinitionSearchFunction defSearch =
```

Comment:
Is this worth being a lambda?

Reply:
It encompasses the notation of giving a "callback"-esque logic for `TypeSpecialMemberPrinter`. Otherwise, there would be a bigger coupling on `TypeSpecialMemberPrinter`'s translation unit onto `CppService`'s interface, and such. This is already a very "hacky" sort of coupling, but the loosest which still provides the functionality without changing the layout of *the entirety of CodeCompass*.